padrino-core/lib/padrino-core/cli/rake_tasks.rb

```diff
@@ -70,3 +62,11 @@ def env_migration_version
     list_app_routes(app, args) if app
   end
 end
+
+Dir["{lib/tasks/**,tasks/**,test,spec}/*.rake"].each do |file|
```

Comment:
> These tasks need to be visible without environment manipulation for discovery reasons.

Discoverable by who and for what? 

> rake -T should always show all tasks, independent of environment variables.

If gem X is included in development and has a task. When I deploy via `bundle install --deployment --without development test` That task is not available via `rake -T`. How is this case different?